### PR TITLE
Add telemetry archive service

### PR DESCRIPTION
**Pull request by Michael-AirMap**

### CODE REVIEW

File: Source/Telemetry/AirMap+Telemetry.swift [hunk 1]
```diff
@@ -29,7 +29,19 @@ extension AirMap {
 		case invalidCredentials
 		case invalidFlight
 	}
-	
+
+	/// Queries archived telemetry for a given flight
+	///	/// - Parameters:
```
Comment (adolfo) [inline Source/Telemetry/AirMap+Telemetry.swift]: Comment alignment looks off

File: Source/Telemetry/Models/SampleRate.swift [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public enum SampleRate {
+	case miliseconds(Int)
```
Comment (adolfo) [inline Source/Telemetry/Models/SampleRate.swift]: *milliseconds

File: Source/Telemetry/Models/SampleRate.swift [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	case minutes(Int)
+	
+	var param: String {
+		switch self {
+		case .miliseconds(let val):
+			return "\(val)ms"
+		case .seconds(let val):
+			return "\(val)s"
+		case .minutes(let val):
+			return "\(val)min"
```
Comment (adolfo) [inline Source/Telemetry/Models/SampleRate.swift]: ~min~ m

File: Source/Telemetry/Models/SampleRate.swift [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+//
+
+import Foundation
+
+public enum SampleRate {
+	case miliseconds(Int)
+	case seconds(Int)
+	case minutes(Int)
+	
+	var param: String {
```
Comment (adolfo) [inline Source/Telemetry/Models/SampleRate.swift]: `CustomStringConvertible` instead?

```var description: String ...```

File: Source/Core/Constants.swift [hunk 5]
```diff
@@ -96,9 +99,9 @@ struct Constants {
 		}
 		
 		struct SampleRate {
```
Comment (adolfo) [inline Source/Core/Constants.swift]: `struct SampleRate: CustomStringConvertible`

File: Source/Telemetry/Clients/ArchiveClient.swift [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+	func queryFlightTelemetry(for flightId: AirMapFlightId, from: Date?, to: Date?, sampleRate: SampleRate?) -> Observable<ArchivedTelemetry> {
+
+		AirMap.logger.debug("Query Flight Telemetry", flightId as Any, from as Any, to as Any, sampleRate as Any)
+
+		var params = [String : Any]()
+		params["flight_id"] = flightId.rawValue
+		params["start"] = from?.iso8601String()
+		params["end"] = to?.iso8601String()
+		params["rate"] = sampleRate?.param
```
Comment (adolfo) [inline Source/Telemetry/Clients/ArchiveClient.swift]: `sampleRate.description` or might be ok with just `sampleRate`...

File: Source/Core/Constants.swift (hunk 5 above)
Comment (Michael-AirMap) [reply Source/Core/Constants.swift]: This is the wrong SampleRate. Which suggests we might want to rename the SampleRate used for the telemetry archive found here

https://github.com/airmap/AirMapSDK-Swift/blob/7f30d1b49f0224f8a2442ff726d60e601b580dfa/Source/Telemetry/Models/SampleRate.swift#L23

File: Source/Telemetry/Clients/ArchiveClient.swift (hunk 6 above)
Comment (Michael-AirMap) [reply Source/Telemetry/Clients/ArchiveClient.swift]: Fixed

**Pull request merged**